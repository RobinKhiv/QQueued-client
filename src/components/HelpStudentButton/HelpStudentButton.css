.helpStudentButton:hover {
  cursor: pointer;
  background: rgb(228, 248, 255);
  color: black;
  border: 1px solid black;
}
