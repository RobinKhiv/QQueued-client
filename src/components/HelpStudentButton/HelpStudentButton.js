import React, { useContext } from "react";
import QueueContext from "../../context/QueueContext";
import Button from "../../components/Button/Button";
import slackService from "../../services/slack-api-service"
import uuidv4 from "uuid";

export default function HelpStudentButton(props) {
  const context = useContext(QueueContext);
  const { mentorName } = props;
  const { history } = props;
  
  const handleOnClick = () => {
    const uuid = uuidv4()
    if(context.queueList.length > 0){
    context.queueList[0].mentorName = mentorName
    context.queueList[0].url = `/room/${uuid}`
    if(context.queueList[0].slack_user_id) {
      const text = `Your mentor is ready go at http://localhost:3000${uuid}`;
      slackService.openDmAndMessage(context.queueList[0].slack_user_id, text)
      }
      
    context.tellUser(context.queueList[0])
    context.helpStudent(mentorName, history, context.queueList[0].id, uuid);
  }
  };

  return <Button onClick={() => handleOnClick()}>Help a student</Button>;
}
