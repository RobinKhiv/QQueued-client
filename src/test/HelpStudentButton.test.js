import React from "react";
import TestRenderer from "react-test-renderer";
import HelpStudentButton from '../components/HelpStudentButton/HelpStudentButton';
import ReactDOM from 'react-dom';

describe("HelpStudentButton function component", () => {
  it.skip("renders without crashing", () => {});

  const div = document.createElement("div");
    ReactDOM.render(<HelpStudentButton />, div);
    ReactDOM.unmountComponentAtNode(div);
  
  it('renders the UI as expected', () => {
    const tree = TestRenderer.create(<HelpStudentButton />).toJSON()
    expect(tree).toMatchSnapshot()
  })
});