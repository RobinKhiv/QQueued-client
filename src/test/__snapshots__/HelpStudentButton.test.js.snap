// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<button
  className="Button"
  onClick={[Function]}
>
  Help a student
</button>
`;
